### ISSUE TITLE
Unable to login

### ISSUE THREAD

**Issue by NViviers** [Snazzah/stub#56]
When running the provided docker container with valid environment variables, and then visiting `localhost:3000/login` I am able to see the login page. However, entering my email, submitting and then clicking on the "Sign in" button in the received email, I simply get sent back to the login page again.

This repeats no matter what I do or change, any help would be appreciated.

**Comment by Snazzah** [comment]
Not sure how this isn't working, but I've mainly used Discord OAuth in development. Email has worked on instances before, just don't know why it doesn't work locally.

**Comment by NViviers** [comment]
Now that I've gotten Discord auth working, I'm able to say for sure that the other auth methods have the same problem. They all seem to do the login process but then send me back to the login page

**Comment by Snazzah** [comment]
Huh weird, I've tested e-mail and Discord in my own environment and they work fine. What environment are you testing this on?

**Comment by himbolion** [comment]
on my setup i have the admin panel on `admin.lio.tld` and the actual shortener on `lio.tld` and everything has been working fine for me


Do you have the Callback URL for NextAuth set to your Admin Panel?
These are mine and it works fine
![image](https://user-images.githubusercontent.com/20520193/197306673-590e21fa-c96d-4c0b-848d-6f58343321cc.png)

**Comment by Codeize** [comment]
I'm also facing this issue.

**Comment by himbolion** [comment]
Could you two share your two env entiries I showed above, here?

**Comment by Codeize** [comment]
> Could you two share your two env entiries I showed above, here?

Sure, they are the same as provided by default, however.
```env
# The callback URL for authentication purposes (ex. "https://example.com")
NEXTAUTH_URL=http://localhost:3000

# The domain of the app
APP_HOSTNAME=localhost:3000
```

**Comment by Snazzah** [comment]
Did you guys fill in the `NEXTAUTH_SECRET`? That would also impact authentication.

Also @himbolion the `APP_HOSTNAME` should've been the same domain as the admin instance (which is redundant from `NEXTAUTH_URL` but whatever) It's a little important since that domain gets assigned when using cookies.

**Comment by NViviers** [comment]
I've got the exact same entries as Codeize, I have filled in `NEXTAUTH_SECRET` through the link provided in the line above that entry.

**Comment by Codeize** [comment]
> I've got the exact same entries as Codeize, I have filled in `NEXTAUTH_SECRET` through the link provided in the line above that entry.

Same here, I did what was asked of me in the example env file.

**Comment by NViviers** [comment]
> Huh weird, I've tested e-mail and Discord in my own environment and they work fine. What environment are you testing this on?

Seems I never answered this. I'm running Pop!_OS here with Docker 20.10.20 and I'm visiting the project in Firefox

**Comment by NViviers** [comment]
I can try running this on Windows as well as Ubuntu, although Ubuntu will be running ARM instead of x86

**Comment by himbolion** [comment]
Weird, it's been running well for me on my Debian Server through Docker, with all those entries filled as well.

Are you using the latest Docker Image, if running through docker?
Or are you using the latest version of the Repo if you've just cloned the Repo?

**Comment by NViviers** [comment]
Yes, I am running the latest Docker image provided

**Comment by Snazzah** [comment]
Well the only thing I can think of is if there were errors while next-auth was handling authentication. Maybe the redirect URI was mistyped when using the Discord OAuth. Does the query string of the login URL change at all? Maybe something like `error=LoginFailed` or something.

**Comment by NViviers** [comment]
I've triple checked the OAuth redirect URI, it's definitely correct. The query string doesn't change. `localhost:3000` sends me to `localhost:3000/login`, after logging in through Discord I get sent back to exactly `localhost:3000/login` again

**Comment by Snazzah** [comment]
Huh, I guess browser settings could impact it? I don't even know, I have the same environment variables and it works fine over localhost and on a separate server.

**Comment by NViviers** [comment]
I've tried on both Firefox and Chrome, both have the same result. Simply send me back to the login page

**Comment by NViviers** [comment]
> Huh, I guess browser settings could impact it? I don't even know, I have the same environment variables and it works fine over localhost and on a separate server.

Are you able to send us your .env file? I want to compare to see what difference there is, we're running through docker so there shouldn't be any problems there. I've tried both Chrome and Firefox (Don't think the problem is there either)

**Comment by himbolion** [comment]
I can grab mine in a bit, since I recently moved servers and it still works fine for me on the new server

**Comment by JackBailey** [comment]
Same issue here

Using docker compose on Ubuntu 20, didn't change anything but env variables. Tried with both magic link (didn't receive email) and github (redirected back to login)

**Comment by himbolion** [comment]
My env looks like this
![image](https://user-images.githubusercontent.com/20520193/206806400-9d6479ae-9e2d-4a4c-b14e-0aeb12f4b07b.png)

my compose.yml looks like this 
![image](https://user-images.githubusercontent.com/20520193/206806543-5e8d0b65-1553-43c3-a119-7252c51bb2d5.png)

my secrets are censored , for obvious reasons. But this has been working fine on my old and now my new server

**Comment by NViviers** [comment]
This might a stupid question, but in the logs I see both ports `3000` and `3001` open, I'm visiting `localhost:3000/login` for the login, I've tried with Discord authentication to redirect to both `localhost:3000/api/auth/callback/discord` and `localhost:3001`, the first one gives me a `Invalid OAuth2 redirect_uri : You can now close this tab.` error message from Discord (Without it redirecting me or anything yet), the second actually redirects me to `http://localhost:3001/api/auth/callback/discord?code=...&state=...` and displays `Not Found`. Am I using this incorrectly?

**Comment by Snazzah** [comment]
Yeah, that's the problem, the redirect URI port needs to be at `3000` since that's where the admin panel is at. You need to add `localhost:3000/api/auth/callback/discord` as a redirect URI in your Discord app page.

**Comment by ggtylerr** [comment]
Trying to use this app and hit the same roadblocks @NViviers is having. Unfortunately setting it as a different port (or really any port) makes Discord / GH not accept the Callback URI (although that might be because I'm running it over a domain and not localhost.)

Additionally, emails don't seem to send. I'm using a poste.io setup and I believe that since it allows multiple domains the SMTP URI doesn't work at all. But that's a nodemailer / next-auth.js issue and I don't think much can be done about it sadly.

**Comment by vdeville** [comment]
Hello,
Do you solve your problem ? Because i have exactly the same issue.
After click on the link received by mail, i will redirect to /login again and again.

I set super-admin from cli after i installed the yarn package for dev, but no change.
```
/app $ yarn manage set-superadmin
yarn run v1.22.19
$ tsx scripts/cli.ts set-superadmin
Enter the email of the user you want to set as a superadmin.
✔ E-mail · valentin.deville@xx.fr
Updating superadmin user ... OK
Done in 6.72s.

```

Do you have any idea to solve the problem ?

Thanks

**Comment by vdeville** [comment]
Same issue with the Github OAuth, redirect to /login again and again.
Thanks @Snazzah

**Comment by robinhalet** [comment]
Has anyone managed to find a solution?

**Comment by marcoantonio123456** [comment]
I dealt with the same issue with SMTP but got github to work no problem. edited user through database and done
